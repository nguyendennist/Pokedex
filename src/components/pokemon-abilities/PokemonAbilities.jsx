import React from 'react';
import PropTypes from 'prop-types';
import PokemonAbility from 'components/pokemon-ability/PokemonAbility.jsx';

const PokemonAbilities = ({abilities, style}) => (
  <div className='pokemon-abilities' style={style}>
    <div>
      <p>{abilities.length > 1 ? 'Abilities' : 'Ability'}</p>
      {
        abilities.map(({ability}) => (
          <PokemonAbility key={ability.url} ability={ability} />
        ))
      }
    </div>
  </div>
);

PokemonAbilities.propTypes = {
  abilities: PropTypes.array.isRequired,
  style: PropTypes.object.isRequired
};

export default PokemonAbilities;
