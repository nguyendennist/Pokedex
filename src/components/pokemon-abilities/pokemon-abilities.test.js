import React from 'react';
import { create } from 'react-test-renderer';
import PokemonAbilities from './PokemonAbilities.jsx';

describe('unit tests', () => {
  test('should be defined', () => {
    expect(PokemonAbilities).toBeDefined();
  });
});

describe('ui tests', () => {
  test('should render correctly', () => {
    const tree = create(
      <PokemonAbilities style={{}} abilities={[]} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
