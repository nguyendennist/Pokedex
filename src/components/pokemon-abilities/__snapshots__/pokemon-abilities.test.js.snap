// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui tests should render correctly 1`] = `
<div
  className="pokemon-abilities"
  style={Object {}}
>
  <div>
    <p>
      Ability
    </p>
  </div>
</div>
`;
